src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2568,13 +2590,14 @@ private TypeSymbol NominalSlotType(int slot)
         /// </summary>
         private void SetStateAndTrackForFinally(ref LocalState state, int slot, NullableFlowState newState)
         {
-            state[slot] = newState;
+            Debug.Assert(slot > 0);
+            SetSlot(ref state, slot, newState);
             if (newState != NullableFlowState.NotNull && NonMonotonicState.HasValue)
             {
                 var tryState = NonMonotonicState.Value;
                 if (tryState.HasVariable(slot))
                 {
-                    tryState[slot] = newState.Join(tryState[slot]);
+                    SetSlot(ref tryState, slot, newState.Join(GetSlot(ref tryState, slot)), useNotNullsAsDefault: true);
```

Comment:
>useNotNullsAsDefault: true [](http://example.com/codeflow?start=92&length=26)

:memo: I couldn't find or make a test where this matters, but I think it's correct given that the nullable walker's `NonMonotonicState` is initialized with `ReachableBottomState` (ie. explicitly filled with `NotNull` by default).